diff --git a/cmake-modules/Findassimp.cmake b/cmake-modules/Findassimp.cmake
index 6c323ada..db98c00a 100644
--- a/cmake-modules/Findassimp.cmake
+++ b/cmake-modules/Findassimp.cmake
@@ -3,7 +3,7 @@ if(CMAKE_SIZEOF_VOID_P EQUAL 8)
 elseif(CMAKE_SIZEOF_VOID_P EQUAL 4)
 	set(ASSIMP_ARCHITECTURE "32")
 endif(CMAKE_SIZEOF_VOID_P EQUAL 8)
-	
+
 if(WIN32)
 	set(ASSIMP_ROOT_DIR CACHE PATH "ASSIMP root directory")
 
@@ -18,12 +18,12 @@ if(WIN32)
 
 	if(MSVC12)
 		set(ASSIMP_MSVC_VERSIONS "vc120")
-	elseif(MSVC14)	
+	elseif(MSVC14)
 		set(ASSIMP_MSVC_VERSIONS "vc140;vc141")
 	endif(MSVC12)
-	
+
 	if(MSVC12 OR MSVC14)
-	
+
     foreach(ASSIMP_MSVC_VERSION ${ASSIMP_MSVC_VERSIONS})
       find_path(ASSIMP_LIBRARY_DIR
         NAMES
@@ -32,27 +32,27 @@ if(WIN32)
         HINTS
           ${ASSIMP_ROOT_DIR}/lib${ASSIMP_ARCHITECTURE}
       )
-      
+
       find_library(ASSIMP_LIBRARY_RELEASE				assimp-${ASSIMP_MSVC_VERSION}-mt.lib 			PATHS ${ASSIMP_LIBRARY_DIR})
       find_library(ASSIMP_LIBRARY_DEBUG				assimp-${ASSIMP_MSVC_VERSION}-mtd.lib			PATHS ${ASSIMP_LIBRARY_DIR})
 
       IF(NOT ASSIMP_LIBRARY_RELEASE AND NOT ASSIMP_LIBRARY_DEBUG)
         continue()
       ENDIF()
-      
+
       IF(ASSIMP_LIBRARY_DEBUG)
-        set(ASSIMP_LIBRARY 
+        set(ASSIMP_LIBRARY
           optimized 	${ASSIMP_LIBRARY_RELEASE}
           debug		${ASSIMP_LIBRARY_DEBUG}
         )
       ELSE()
-        set(ASSIMP_LIBRARY 
+        set(ASSIMP_LIBRARY
           optimized 	${ASSIMP_LIBRARY_RELEASE}
         )
       ENDIF()
-      
+
       set(ASSIMP_LIBRARIES ${ASSIMP_LIBRARY_RELEASE} ${ASSIMP_LIBRARY_DEBUG})
-    
+
       FUNCTION(ASSIMP_COPY_BINARIES TargetDirectory)
         ADD_CUSTOM_TARGET(AssimpCopyBinaries
           COMMAND ${CMAKE_COMMAND} -E copy ${ASSIMP_ROOT_DIR}/bin${ASSIMP_ARCHITECTURE}/assimp-${ASSIMP_MSVC_VERSION}-mtd.dll 	${TargetDirectory}/Debug/assimp-${ASSIMP_MSVC_VERSION}-mtd.dll
@@ -60,11 +60,11 @@ if(WIN32)
         COMMENT "Copying Assimp binaries to '${TargetDirectory}'"
         VERBATIM)
       ENDFUNCTION(ASSIMP_COPY_BINARIES)
-    
+
       SET(assimp_LIBRARIES ${ASSIMP_LIBRARY})
     endforeach()
 	endif()
-	
+
 else(WIN32)
 
 	find_path(
@@ -96,5 +96,5 @@ else(WIN32)
 		message(FATAL_ERROR "Could not find asset importer library")
 	  endif (assimp_FIND_REQUIRED)
 	endif (assimp_FOUND)
-	
+
 endif(WIN32)
