diff --git a/cmake-modules/Findassimp.cmake b/cmake-modules/Findassimp.cmake
index 8b7cefd0..0b3dba21 100644
--- a/cmake-modules/Findassimp.cmake
+++ b/cmake-modules/Findassimp.cmake
@@ -4,8 +4,8 @@ elseif(CMAKE_SIZEOF_VOID_P EQUAL 4)
 	set(ASSIMP_ARCHITECTURE "32")
 endif(CMAKE_SIZEOF_VOID_P EQUAL 8)
 
+set(ASSIMP_ROOT_DIR CACHE PATH "ASSIMP root directory")
 if(WIN32)
-	set(ASSIMP_ROOT_DIR CACHE PATH "ASSIMP root directory")
 
 	# Find path of each library
 	find_path(ASSIMP_INCLUDE_DIR
@@ -14,7 +14,7 @@ if(WIN32)
 		HINTS
 			${ASSIMP_ROOT_DIR}/include
 	)
-  SET(assimp_INCLUDE_DIRS ${ASSIMP_INCLUDE_DIR})
+  	SET(assimp_INCLUDE_DIRS ${ASSIMP_INCLUDE_DIR})
 
 	if(MSVC12)
 		set(ASSIMP_MSVC_VERSIONS "vc120")
@@ -24,45 +24,53 @@ if(WIN32)
 
 	if(MSVC)
 
-    foreach(ASSIMP_MSVC_VERSION ${ASSIMP_MSVC_VERSIONS})
-      find_path(ASSIMP_LIBRARY_DIR
-        NAMES
-          assimp-${ASSIMP_MSVC_VERSION}-mt.lib
-          assimp-${ASSIMP_MSVC_VERSION}-mtd.lib
-        HINTS
-          ${ASSIMP_ROOT_DIR}/lib${ASSIMP_ARCHITECTURE}
-      )
-
-      find_library(ASSIMP_LIBRARY_RELEASE				assimp-${ASSIMP_MSVC_VERSION}-mt.lib 			PATHS ${ASSIMP_LIBRARY_DIR})
-      find_library(ASSIMP_LIBRARY_DEBUG				assimp-${ASSIMP_MSVC_VERSION}-mtd.lib			PATHS ${ASSIMP_LIBRARY_DIR})
-
-      IF(NOT ASSIMP_LIBRARY_RELEASE AND NOT ASSIMP_LIBRARY_DEBUG)
-        continue()
-      ENDIF()
-
-      IF(ASSIMP_LIBRARY_DEBUG)
-        set(ASSIMP_LIBRARY
-          optimized 	${ASSIMP_LIBRARY_RELEASE}
-          debug		${ASSIMP_LIBRARY_DEBUG}
-        )
-      ELSE()
-        set(ASSIMP_LIBRARY
-          optimized 	${ASSIMP_LIBRARY_RELEASE}
-        )
-      ENDIF()
-
-      set(ASSIMP_LIBRARIES ${ASSIMP_LIBRARY_RELEASE} ${ASSIMP_LIBRARY_DEBUG})
-
-      FUNCTION(ASSIMP_COPY_BINARIES TargetDirectory)
-        ADD_CUSTOM_TARGET(AssimpCopyBinaries
-          COMMAND ${CMAKE_COMMAND} -E copy ${ASSIMP_ROOT_DIR}/bin${ASSIMP_ARCHITECTURE}/assimp-${ASSIMP_MSVC_VERSION}-mtd.dll 	${TargetDirectory}/Debug/assimp-${ASSIMP_MSVC_VERSION}-mtd.dll
-          COMMAND ${CMAKE_COMMAND} -E copy ${ASSIMP_ROOT_DIR}/bin${ASSIMP_ARCHITECTURE}/assimp-${ASSIMP_MSVC_VERSION}-mt.dll 		${TargetDirectory}/Release/assimp-${ASSIMP_MSVC_VERSION}-mt.dll
-        COMMENT "Copying Assimp binaries to '${TargetDirectory}'"
-        VERBATIM)
-      ENDFUNCTION(ASSIMP_COPY_BINARIES)
-
-      SET(assimp_LIBRARIES ${ASSIMP_LIBRARY})
-    endforeach()
+		find_path(ASSIMP_LIBRARY_DIR
+			NAMES
+			assimp.lib
+			assimpd.lib
+			HINTS
+			${ASSIMP_ROOT_DIR}/lib${ASSIMP_ARCHITECTURE}
+		)
+
+		find_library(ASSIMP_LIBRARY_RELEASE assimp.lib PATHS ${ASSIMP_LIBRARY_DIR})
+		find_library(ASSIMP_LIBRARY_DEBUG assimpd.lib PATHS ${ASSIMP_LIBRARY_DIR})
+
+		IF(NOT ASSIMP_LIBRARY_RELEASE AND NOT ASSIMP_LIBRARY_DEBUG)
+			foreach(ASSIMP_MSVC_VERSION ${ASSIMP_MSVC_VERSIONS})
+				find_path(ASSIMP_LIBRARY_DIR
+					NAMES
+					assimp-${ASSIMP_MSVC_VERSION}-mt.lib
+					assimp-${ASSIMP_MSVC_VERSION}-mtd.lib
+					HINTS
+					${ASSIMP_ROOT_DIR}/lib${ASSIMP_ARCHITECTURE}
+				)
+
+				find_library(ASSIMP_LIBRARY_RELEASE				assimp-${ASSIMP_MSVC_VERSION}-mt.lib 			PATHS ${ASSIMP_LIBRARY_DIR})
+				find_library(ASSIMP_LIBRARY_DEBUG				assimp-${ASSIMP_MSVC_VERSION}-mtd.lib			PATHS ${ASSIMP_LIBRARY_DIR})
+
+				IF(ASSIMP_LIBRARY_RELEASE OR ASSIMP_LIBRARY_DEBUG)
+					break()
+				ENDIF()
+			endforeach()
+		ENDIF(NOT ASSIMP_LIBRARY_RELEASE AND NOT ASSIMP_LIBRARY_DEBUG)
+
+		IF(NOT ASSIMP_LIBRARY_RELEASE AND NOT ASSIMP_LIBRARY_DEBUG)
+			SET(assimp_FOUND FALSE)
+			return()
+		ENDIF()
+
+		IF(ASSIMP_LIBRARY_DEBUG)
+			set(ASSIMP_LIBRARY
+			optimized 	${ASSIMP_LIBRARY_RELEASE}
+			debug		${ASSIMP_LIBRARY_DEBUG}
+			)
+		ELSE()
+			set(ASSIMP_LIBRARY
+			optimized 	${ASSIMP_LIBRARY_RELEASE}
+			)
+		ENDIF()
+
+		SET(assimp_LIBRARIES ${ASSIMP_LIBRARY})
 	endif()
 
 else(WIN32)
@@ -72,7 +80,7 @@ else(WIN32)
 	  NAMES assimp/postprocess.h assimp/scene.h assimp/version.h assimp/config.h assimp/cimport.h
 	  PATHS /usr/local/include
 	  PATHS /usr/include/
-
+		HINTS	${ASSIMP_ROOT_DIR}/include
 	)
 
 	find_library(
