From b377becd5625a398118e371ba762936a52089fbe Mon Sep 17 00:00:00 2001
From: Justin Carpentier <justin.carpentier@inria.fr>
Date: Fri, 7 Aug 2020 13:35:13 +0200
Subject: [PATCH 1/2] cmake: fix search of assimp on Windows

---
 cmake-modules/Findassimp.cmake | 58 ++++++++++++++++++----------------
 1 file changed, 30 insertions(+), 28 deletions(-)

diff --git a/cmake-modules/Findassimp.cmake b/cmake-modules/Findassimp.cmake
index fcbdd0d..9963180 100644
--- a/cmake-modules/Findassimp.cmake
+++ b/cmake-modules/Findassimp.cmake
@@ -17,39 +17,41 @@ if(WIN32)
   SET(assimp_INCLUDE_DIRS ${ASSIMP_INCLUDE_DIR})
 
 	if(MSVC12)
-		set(ASSIMP_MSVC_VERSION "vc120")
+		set(ASSIMP_MSVC_VERSIONS "vc120")
 	elseif(MSVC14)	
-		set(ASSIMP_MSVC_VERSION "vc140")
+		set(ASSIMP_MSVC_VERSIONS "vc140;vc141")
 	endif(MSVC12)
 	
 	if(MSVC12 OR MSVC14)
 	
-		find_path(ASSIMP_LIBRARY_DIR
-			NAMES
-				assimp-${ASSIMP_MSVC_VERSION}-mt.lib
-			HINTS
-				${ASSIMP_ROOT_DIR}/lib${ASSIMP_ARCHITECTURE}
-		)
-		
-		find_library(ASSIMP_LIBRARY_RELEASE				assimp-${ASSIMP_MSVC_VERSION}-mt.lib 			PATHS ${ASSIMP_LIBRARY_DIR})
-		find_library(ASSIMP_LIBRARY_DEBUG				assimp-${ASSIMP_MSVC_VERSION}-mtd.lib			PATHS ${ASSIMP_LIBRARY_DIR})
-		
-		set(ASSIMP_LIBRARY 
-			optimized 	${ASSIMP_LIBRARY_RELEASE}
-			debug		${ASSIMP_LIBRARY_DEBUG}
-		)
-		
-		set(ASSIMP_LIBRARIES ${ASSIMP_LIBRARY_RELEASE} ${ASSIMP_LIBRARY_DEBUG})
-	
-		FUNCTION(ASSIMP_COPY_BINARIES TargetDirectory)
-			ADD_CUSTOM_TARGET(AssimpCopyBinaries
-				COMMAND ${CMAKE_COMMAND} -E copy ${ASSIMP_ROOT_DIR}/bin${ASSIMP_ARCHITECTURE}/assimp-${ASSIMP_MSVC_VERSION}-mtd.dll 	${TargetDirectory}/Debug/assimp-${ASSIMP_MSVC_VERSION}-mtd.dll
-				COMMAND ${CMAKE_COMMAND} -E copy ${ASSIMP_ROOT_DIR}/bin${ASSIMP_ARCHITECTURE}/assimp-${ASSIMP_MSVC_VERSION}-mt.dll 		${TargetDirectory}/Release/assimp-${ASSIMP_MSVC_VERSION}-mt.dll
-			COMMENT "Copying Assimp binaries to '${TargetDirectory}'"
-			VERBATIM)
-		ENDFUNCTION(ASSIMP_COPY_BINARIES)
-	
-    SET(assimp_LIBRARIES ${ASSIMP_LIBRARY_RELEASE})
+    foreach(ASSIMP_MSVC_VERSION ${ASSIMP_MSVC_VERSIONS})
+      find_path(ASSIMP_LIBRARY_DIR
+        NAMES
+          assimp-${ASSIMP_MSVC_VERSION}-mt.lib
+        HINTS
+          ${ASSIMP_ROOT_DIR}/lib${ASSIMP_ARCHITECTURE}
+      )
+      
+      find_library(ASSIMP_LIBRARY_RELEASE				assimp-${ASSIMP_MSVC_VERSION}-mt.lib 			PATHS ${ASSIMP_LIBRARY_DIR})
+      find_library(ASSIMP_LIBRARY_DEBUG				assimp-${ASSIMP_MSVC_VERSION}-mtd.lib			PATHS ${ASSIMP_LIBRARY_DIR})
+      
+      set(ASSIMP_LIBRARY 
+        optimized 	${ASSIMP_LIBRARY_RELEASE}
+        debug		${ASSIMP_LIBRARY_DEBUG}
+      )
+      
+      set(ASSIMP_LIBRARIES ${ASSIMP_LIBRARY_RELEASE} ${ASSIMP_LIBRARY_DEBUG})
+    
+      FUNCTION(ASSIMP_COPY_BINARIES TargetDirectory)
+        ADD_CUSTOM_TARGET(AssimpCopyBinaries
+          COMMAND ${CMAKE_COMMAND} -E copy ${ASSIMP_ROOT_DIR}/bin${ASSIMP_ARCHITECTURE}/assimp-${ASSIMP_MSVC_VERSION}-mtd.dll 	${TargetDirectory}/Debug/assimp-${ASSIMP_MSVC_VERSION}-mtd.dll
+          COMMAND ${CMAKE_COMMAND} -E copy ${ASSIMP_ROOT_DIR}/bin${ASSIMP_ARCHITECTURE}/assimp-${ASSIMP_MSVC_VERSION}-mt.dll 		${TargetDirectory}/Release/assimp-${ASSIMP_MSVC_VERSION}-mt.dll
+        COMMENT "Copying Assimp binaries to '${TargetDirectory}'"
+        VERBATIM)
+      ENDFUNCTION(ASSIMP_COPY_BINARIES)
+    
+      SET(assimp_LIBRARIES ${ASSIMP_LIBRARY_RELEASE})
+    endforeach()
 	endif()
 	
 else(WIN32)
-- 
2.21.1 (Apple Git-122.3)

