From b5e447c9dfb2bd5a84d23882026a22c978b0a5d6 Mon Sep 17 00:00:00 2001
From: Justin Carpentier <justin.carpentier@inria.fr>
Date: Fri, 7 Aug 2020 13:45:24 +0200
Subject: [PATCH 2/2] cmake: fix check

---
 cmake-modules/Findassimp.cmake | 21 ++++++++++++++++-----
 1 file changed, 16 insertions(+), 5 deletions(-)

diff --git a/cmake-modules/Findassimp.cmake b/cmake-modules/Findassimp.cmake
index 9963180..6c323ad 100644
--- a/cmake-modules/Findassimp.cmake
+++ b/cmake-modules/Findassimp.cmake
@@ -28,17 +28,28 @@ if(WIN32)
       find_path(ASSIMP_LIBRARY_DIR
         NAMES
           assimp-${ASSIMP_MSVC_VERSION}-mt.lib
+          assimp-${ASSIMP_MSVC_VERSION}-mtd.lib
         HINTS
           ${ASSIMP_ROOT_DIR}/lib${ASSIMP_ARCHITECTURE}
       )
       
       find_library(ASSIMP_LIBRARY_RELEASE				assimp-${ASSIMP_MSVC_VERSION}-mt.lib 			PATHS ${ASSIMP_LIBRARY_DIR})
       find_library(ASSIMP_LIBRARY_DEBUG				assimp-${ASSIMP_MSVC_VERSION}-mtd.lib			PATHS ${ASSIMP_LIBRARY_DIR})
+
+      IF(NOT ASSIMP_LIBRARY_RELEASE AND NOT ASSIMP_LIBRARY_DEBUG)
+        continue()
+      ENDIF()
       
-      set(ASSIMP_LIBRARY 
-        optimized 	${ASSIMP_LIBRARY_RELEASE}
-        debug		${ASSIMP_LIBRARY_DEBUG}
-      )
+      IF(ASSIMP_LIBRARY_DEBUG)
+        set(ASSIMP_LIBRARY 
+          optimized 	${ASSIMP_LIBRARY_RELEASE}
+          debug		${ASSIMP_LIBRARY_DEBUG}
+        )
+      ELSE()
+        set(ASSIMP_LIBRARY 
+          optimized 	${ASSIMP_LIBRARY_RELEASE}
+        )
+      ENDIF()
       
       set(ASSIMP_LIBRARIES ${ASSIMP_LIBRARY_RELEASE} ${ASSIMP_LIBRARY_DEBUG})
     
@@ -50,7 +61,7 @@ if(WIN32)
         VERBATIM)
       ENDFUNCTION(ASSIMP_COPY_BINARIES)
     
-      SET(assimp_LIBRARIES ${ASSIMP_LIBRARY_RELEASE})
+      SET(assimp_LIBRARIES ${ASSIMP_LIBRARY})
     endforeach()
 	endif()
 	
-- 
2.21.1 (Apple Git-122.3)

