diff --git a/cmake-modules/Findassimp.cmake b/cmake-modules/Findassimp.cmake
index db98c00a..8b7cefd0 100644
--- a/cmake-modules/Findassimp.cmake
+++ b/cmake-modules/Findassimp.cmake
@@ -18,11 +18,11 @@ if(WIN32)
 
 	if(MSVC12)
 		set(ASSIMP_MSVC_VERSIONS "vc120")
-	elseif(MSVC14)
-		set(ASSIMP_MSVC_VERSIONS "vc140;vc141")
+	else()
+		set(ASSIMP_MSVC_VERSIONS "vc140;vc141;vc142;vc143")
 	endif(MSVC12)
 
-	if(MSVC12 OR MSVC14)
+	if(MSVC)
 
     foreach(ASSIMP_MSVC_VERSION ${ASSIMP_MSVC_VERSIONS})
       find_path(ASSIMP_LIBRARY_DIR
